Add more upgrading examples (#615)
sous-chefs/apache2: UPGRADING.md

end
```

## Where did the apache2 recipes go?

Recipes are now modules, so your cookbook pattern will go from

```ruby
include_recipe 'apache2'
include_recipe 'apache2::mod_proxy'
include_recipe 'apache2::mod_proxy_http'
include_recipe 'apache2::mod_ssl'
```
to
```ruby
apache2_install 'default_install'
apache2_module 'headers'
apache2_module 'proxy'
apache2_module 'proxy_http'
apache2_module 'rewrite'
apache2_module 'ssl'

> ```

Reviewer: This is awesome 👍

It is recommended that you manage your template and call the `apache2_site` resource directly.

> This leads to a simpler system, and you get more control over the variables and values you pass to the template.  Note that the variables passed to the template have also changed.

Reviewer: ```suggestion
This leads to a simpler system, and you get more control over the variables and values you pass to the template.  

Note that the variables passed to the template have also changed.
```

Reviewer: Adding in a line break here will help readability

```ruby
web_app 'ssl_redirect' do
  server_name node['hostname']
  server_port '80'
  docroot '/var/www/'
  template 'ssl_redirect.conf.erb'
  notifies :restart, 'service[apache2]', :delayed
end
```
to
```ruby
apache2_default_site 'ssl_redirect' do
  default_site_name 'ssl_redirect'
  cookbook 'my_cookbook'
  port '80'
  template_source 'ssl_redirect.conf.erb'
  action :enable

> end

Reviewer: These aren't the same thing and it's misleading to say so here :/

```

> An further example of this behaviour can be seen in the `apache2_default_site` resource.

Reviewer: ```suggestion
A further example of this behaviour can be seen in the `apache2_default_site` resource.
```

> ## One last thing...

Reviewer: ```suggestion
## The service resource
```

> `service['apache2']` is defined in the apache2_default_install resource but other resources are currently unable to reference it.  To work around this issue, define the following helper in your cookbook:

Reviewer: ```suggestion
whilst `service['apache2']` is defined in the `apache2_install` resource, due to the notification system of nested resources, we are unable to notify or subscribe directly to it. 

To work around this issue, define the following helper in your cookbook:
```
